framework/src/main/java/org/checkerframework/framework/ajava/InsertAjavaAnnotations.java

```diff
@@ -0,0 +1,580 @@
+            annotationInputStream.close();
+            Files.write(path, result.getBytes());
+        } catch (IOException e) {
+            System.err.println(
+                    "Failed to insertion annotations from file "
+                            + annotationFilePath
+                            + " to file "
+                            + javaFilePath);
+            System.exit(1);
+        }
+    }
+
+    /**
+     * Inserts annotations from ajava files into Java files in place.
+     *
+     * <p>The first argument is a file or directory containing ajava files. It may be a single ajava
+     * file or a directory containing ajava files.
+     *
+     * <p>The second argument is a Java file or a directory containing Java files to insert
+     * annotations into.
+     *
+     * <p>For each file in the second argument, checks if an ajava file from the first argument
+     * matches it. For each such file, inserts all its annotations into the Java file.
+     *
+     * @param args command line arguments, the first element should be a path to ajava files and the
+     *     second should be the directory containing Java files to insert into
+     */
+    public static void main(String[] args) {
+        if (args.length != 2) {
+            System.out.println(
+                    "Usage: java InsertAjavaAnnotations <ajava-directory> <java-file-directory");
+            System.exit(1);
+        }
+
+        AnnotationFileStore annotationFiles = new AnnotationFileStore();
+        annotationFiles.addFileOrDirectory(new File(args[0]));
+        InsertAjavaAnnotations inserter = new InsertAjavaAnnotations(createElements());
+        FileVisitor<Path> visitor =
+                new SimpleFileVisitor<Path>() {
+                    @Override
+                    public FileVisitResult visitFile(Path path, BasicFileAttributes attrs) {
+                        if (!path.getFileName().toString().endsWith(".java")) {
+                            return FileVisitResult.CONTINUE;
+                        }
+
+                        CompilationUnit root = null;
+                        try {
+                            root = StaticJavaParser.parse(path);
+                        } catch (IOException e) {
+                            System.err.println("Failed to read file: " + path);
+                            System.exit(1);
+                        }
+
+                        Set<String> annotationFilesForRoot = new LinkedHashSet<>();
+                        for (TypeDeclaration<?> type : root.getTypes()) {
+                            String name = type.getNameAsString();
+                            if (root.getPackageDeclaration().isPresent()) {
+                                name =
+                                        root.getPackageDeclaration().get().getNameAsString()
+                                                + "."
+                                                + name;
+                            }
```

Comment:
I suggest abstracting this logic into a method (especially since I think it occurs at least twice).
It could go in file `JavaParserUtils.java`.